packages/terra-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -192,6 +195,12 @@ function WorklistDataGrid(props) {
     // Since the row selection mode has changed, the row selection mode needs to be updated.
     setRowSelectionModeAriaLiveMessage(intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklistDataGrid.row-selection-mode-enabled' : 'Terra.worklistDataGrid.row-selection-mode-disabled' }));
 
+    if (gridReceivedFocus.current) {
+      let newFocusCell = dataGridFuncRef.current.getFocusedCell();
+      newFocusCell = { row: newFocusCell.row, col: Math.max(newFocusCell.col + (hasSelectableRows ? 1 : -1), 0) };
+      dataGridFuncRef.current.setFocusedRowCol(newFocusCell.row, newFocusCell.col, gridFocusedWhenRowSelectionDisabled.current);
+    }
```

Comment:
Do we really need to track focus?  I believe the DataGrid already calls the method to set focus when it is received.  All we would need to do is ensure the proper state is set before focus is received.  That should be able to be handled in the useEffect.